### ISSUE TITLE
_ReindexCatIds is broken

### ISSUE THREAD

**Issue by thovden** [pylabel-project/pylabel#52]
https://github.com/thovden/pylabel/blob/c17d2c644f1dcb85ef3f9835c8771b96b969a673/pylabel/shared.py#L39

The `__ReindexCatIds` function is broken, mainly because it's operating on a copy of a DataFrame. 

The following code does **not** work [in-place](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.replace.html) by default: 

```python
    df = df.replace(r"^\s*$", np.nan, regex=True)
```

Hence, the rest of the function operates on a copy of the DataFrame, but the intention is to do in-place replacements of `df['cat_id']`. 

I also suspect the intention of the following code is probably to coerce `df['cat_id']` to an numeric type: 

```
pd.to_numeric(df["cat_id"])
```

However, these values are not written back to the original data frame. 

I'm going to cleanup this function a little and submit a PR.

**Comment by thovden** [comment]
The issue seems to be introduced by this commit https://github.com/pylabel-project/pylabel/commit/23b28bf7ed696c7221353fd48f82de1bfe647591

**Comment by alexheat** [comment]
Thank you again @thovden . I have created pre-release version 0.1.39a1 with your fix.

Could you give it a try by installing
pip install pylabel==0.1.39a1

**Comment by thovden** [comment]
Verified to work in 0.1.39a1. Thanks, @alexheat!

**Comment by alexheat** [comment]
Thank you again @thovden . I have released 0.1.39 with the fix. 

I also added some test cases to make sure this is not broken again in the future https://github.com/pylabel-project/pylabel/commit/a9762242c77574e804393c8590912fed7c65b9ca 

The test cases run whenever a new PR is made.
